storage/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogManagerConfig.java

```diff
@@ -393,6 +413,7 @@ public boolean equals(Object o) {
                 && Objects.equals(remoteLogMetadataManagerClassName, that.remoteLogMetadataManagerClassName)
                 && Objects.equals(remoteLogMetadataManagerClassPath, that.remoteLogMetadataManagerClassPath)
                 && Objects.equals(remoteLogMetadataManagerListenerName, that.remoteLogMetadataManagerListenerName)
+                && remoteLogMetadataCustomMetadataMaxSize == that.remoteLogMetadataCustomMetadataMaxSize
```

Comment:
nit: Could we move this line up to L411, since this is a cheaper comparison compared with object comparison.

Reply:
Sure!